src/main/scala/vsys/api/http/TransactionsApiRoute.scala

```diff
@@ -172,15 +172,35 @@ case class TransactionsApiRoute(
     pathPrefix(Segment) { address =>
       Address.fromString(address) match {
         case Left(e) => complete(ApiError.fromValidationError(e))
-        case Right(a) =>
-          complete(Json.arr(JsArray(state.activeLeases().flatMap(state.transactionInfo)
+        case Right(a) => {
+          if(stateSettings.txTypeAccountTxIds){
+            val txNum = state.txTypeAccTxLengths(TransactionType.LeaseTransaction, a)
+            val txIds = state.txTypeAccountTxIds(TransactionType.LeaseTransaction, a, txNum, 0)
+            val res = Option(txIds._2)
+              .map(_.flatMap(state.transactionInfo)
+                .map(a => (a._1,a._2,a._2.transaction))
```

Comment:
do not copy the codes below.  result of `state.activeLeases()` is already filtered by lease status